src/main/java/org/elasticsearch/indices/query/IndicesQueriesModule.java

```diff
@@ -30,30 +30,21 @@
 
 public class IndicesQueriesModule extends AbstractModule {
 
-    private Set<Class<QueryParser>> queryParsersClasses = Sets.newHashSet();
-    private Set<QueryParser> queryParsers = Sets.newHashSet();
+    private Set<Class<? extends QueryParser>> queryParsersClasses = Sets.newHashSet();
 
-    public synchronized IndicesQueriesModule addQuery(Class<QueryParser> queryParser) {
+    public synchronized IndicesQueriesModule addQuery(Class<? extends QueryParser> queryParser) {
```

Comment:
+1 on the <? extends QueryParser> bit. I dislike the generics gymnastics you have to do to register the query without it.